core/block_svg.ts

```diff
@@ -996,23 +1008,22 @@ export class BlockSvg
         );
       }
 
-      if (!this.warning) {
-        this.warning = new Warning(this);
+      if (icon) {
+        (icon as WarningIcon).addMessage(text, id);
+      } else {
+        this.addIcon(new WarningIcon(this).addMessage(text, id));
         changedState = true;
       }
-      this.warning!.setText(text, id);
-    } else {
+    } else if (icon) {
       // Dispose all warnings if no ID is given.
-      if (this.warning && !id) {
-        this.warning.dispose();
+      if (!id) {
+        this.removeIcon(WarningIcon.TYPE);
         changedState = true;
-      } else if (this.warning) {
-        const oldText = this.warning.getText();
-        this.warning.setText('', id);
-        const newText = this.warning.getText();
-        if (!newText) {
-          this.warning.dispose();
-        }
+      } else {
+        const oldText = icon.getText();
+        icon.addMessage('', id);
+        const newText = icon.getText();
+        if (!newText) this.removeIcon(WarningIcon.TYPE);
         changedState = oldText !== newText;
       }
     }
```

Comment:
We should check if we can remove the `changedState` stuff since `addIcon` / `removeIcon` should trigger a rerender.

Reply:
Forced rerender was unnecessary!